core/server/data/migrations/versions/4.0/17-solve-orphaned-webhooks.js

```diff
@@ -0,0 +1,47 @@
+const {createIrreversibleMigration} = require('../../utils');
+const logging = require('../../../../../shared/logging');
+const ObjectID = require('bson-objectid');
+
+module.exports = createIrreversibleMigration(
+    async function up(knex) {
+        logging.info('Resolving the orphaned webhooks');
+
+        const orphanedWebhooks = await knex('webhooks')
+            .select('webhooks.id')
+            .leftJoin('integrations', 'integrations.id', 'webhooks.integration_id')
+            .where('integrations.id', null);
+
+        for (let i = 0; i < orphanedWebhooks.length; i++) {
+            // Create new integration
+
+            const webhook = orphanedWebhooks[i];
+            const now = knex.raw('CURRENT_TIMESTAMP');
+
+            const integration = {
+                id: ObjectID.generate(),
+                type: 'custom',
+                name: `Generated integration ${i + 1}`,
+                slug: `generated-integration-${i + 1}`,
```

Comment:
Because of a low but still a possibility to hit `slug` unique constraint using `i` as a number is suboptimal :thinking: A randomly generated string with a number should be less prone to hitting the constraint.

Reply:
Good point, I've switched to the id of the object. Is it ok to share the id of the object? If not I could do another `ObjectID.generate()` for the name/slug.